src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -156,8 +158,13 @@ public async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdentit
 
                 if (result != null)
                 {
-                    return result.WithVersions(asyncValueFactory: 
-                        () => FetchAndMergeVersionsAsync(identity, includePrerelease, cancellationToken));
+                    var versionsAndDeprecationMetadataTask = FetchAndMergeVersionsAndDeprecationMetadataAsync(identity, includePrerelease, cancellationToken);
+
+                    return PackageSearchMetadataBuilder
+                        .FromMetadata(result)
+                        .WithVersions(AsyncLazy.New(async () => (await versionsAndDeprecationMetadataTask).versions))
+                        .WithDeprecation(AsyncLazy.New(async () => (await versionsAndDeprecationMetadataTask).deprecationMetadata))
```

Comment:
Can you create one `AsyncLazy<T>` instance and pass it twice?

Reply:
I don't think so...the properties they grab differ, so there still needs to be some sort of lambda here picking which property to grab. Not sure if there's a better way to write this though.